src/main/scala/chisel3/util/LoadMemoryTransform.scala

```diff
@@ -0,0 +1,174 @@
+// See LICENSE for license details.
+
+package chisel3.util
+
+import chisel3.MemBase
+import chisel3.core.{ChiselAnnotation, RunFirrtlTransform, annotate}
+import chisel3.internal.{Builder, InstanceId}
```

Comment:
Things in package `chisel3.util` should try to be like what any library would look like so it should not reach into `chisel3.core` (import stuff directly from `chisel3_` and `chisel3.experimental._` instead).

Also it should avoid use of `chisel3.internal`, and avoid `private[chisel3]` stuff (like the Builder). I think the only use of Builder is for a warning below that should probably just be an exception, or better yet, a requirement below.

Reply:
Done, I had to add experimental.annotate and chisel3.InstanceId to the chisel package. But I think that is correct as these are part of the public API